### ISSUE TITLE
[BUG] Unable to setup backup-target with error: There is a VM either in the process of creating or deleting

### ISSUE THREAD

**Issue by lanfon72** [harvester/harvester#4545]
### Describe the bug
https://github.com/harvester/harvester/assets/5169694/73ac1dd0-4e34-4d6d-bbfc-8163945a383d

### To Reproduce
1. Install Harvester with any nodes
2. Setup **backup-target**
3. Create an image for VM creation
4. Create VM `vm1` and take backup `vm1b`
5. Restore `vm1b` into new VM `VM1r`
6. Reset **backup-target** to default value
7. Setup **backup-target**

### Expected behavior
We should able to change backup-target, there haven't any VMs. 

## Environment:
 - Harvester ISO version: **v1.2-a9968541-head**
 - **ui-source** Option: **Auto**
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): **Baremetal DL360 4 nodes**

### Additional context
I'm not 100% sure that the root cause is related to #4544, but hit the bug after it.  (so I labeled `reproduce/often` not `reproduce/always`)

### Support bundle
The SB size is too large (>25MB), separated into following:
[harvester-tchfx.zip](https://github.com/harvester/harvester/files/12662679/harvester-tchfx.zip)
[harvester-d5hzw.zip](https://github.com/harvester/harvester/files/12662680/harvester-d5hzw.zip)
[harvester-d2csm.zip](https://github.com/harvester/harvester/files/12662681/harvester-d2csm.zip)
[harvester-gv6lk.zip](https://github.com/harvester/harvester/files/12662682/harvester-gv6lk.zip)
[supportbundle_c6db0211-4e97-404e-bb78-72f4606f5a86_2023-09-19T14-55-03Z.zip](https://github.com/harvester/harvester/files/12662688/supportbundle_c6db0211-4e97-404e-bb78-72f4606f5a86_2023-09-19T14-55-03Z.zip)

**Comment by FrankYang0529** [comment]
After some investigation, I think we may need to prevent users from giving uppercase letters.

```
VirtualMachineRestore encountered error
VirtualMachine.kubevirt.io "VM1r" is invalid: metadata.name: Invalid value: "VM1r": a lowercase RFC 1123 subdomain mus
t consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.
g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

**Comment by lanfon72** [comment]
~~Encountering this **without** restore invalid VM name, the root cause might not be the same as #4544~~

Double confirmed that there were tried to restore VM starts with upper case.

**Comment by lanfon72** [comment]
Confirmed that the bug caused by #4544, so this issue would be closed along with #4544.

For users encountered this, it could be fix by following commands:
```
# list all restores, and find the stuck one
kubectl get virtualmachinerestores.harvesterhci.io -A
# delete it
kubectl delete virtualmachinerestores.harvesterhci.io  -ndefault <THE_STUCK_RESTORE_NAME>
```

**Comment by lanfon72** [comment]
This issue could be closed along with #4552

**Comment by lanfon72** [comment]
Close along with #4552
